_examples/profile/dynamicshare.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/getyoti/yoti-go-sdk/v3"
+	"github.com/getyoti/yoti-go-sdk/v3/dynamic"
+)
+
+func dynamicShare(w http.ResponseWriter, req *http.Request) {
+	policy, err := (&dynamic.PolicyBuilder{}).WithFullName().WithEmail().Build()
+	if err != nil {
+		errorPage(w, req.WithContext(context.WithValue(
+			req.Context(),
+			contextKey("yotiError"),
+			fmt.Sprintf(scenarioBuilderErr, err),
+		)))
+		return
+	}
+	scenario, err := (&dynamic.ScenarioBuilder{}).WithPolicy(policy).WithCallbackEndpoint(profileEndpoint).Build()
+	if err != nil {
+		errorPage(w, req.WithContext(context.WithValue(
+			req.Context(),
+			contextKey("yotiError"),
+			fmt.Sprintf(scenarioBuilderErr, err),
+		)))
+		return
+	}
+
+	pageFromScenario(w, req, "Dynamic Share example", scenario)
+}
+
+func pageFromScenario(w http.ResponseWriter, req *http.Request, title string, scenario dynamic.Scenario) {
+	sdkID := os.Getenv("YOTI_CLIENT_SDK_ID")
+
+	key, err := ioutil.ReadFile(os.Getenv("YOTI_KEY_FILE_PATH"))
+	if err != nil {
+		errorPage(w, req.WithContext(context.WithValue(
+			req.Context(),
+			contextKey("yotiError"),
+			fmt.Sprintf("Unable to retrieve `YOTI_KEY_FILE_PATH`. Error: `%s`", err.Error()),
+		)))
+		return
+	}
+
+	client, err := yoti.NewClient(sdkID, key)
+	if err != nil {
+		errorPage(w, req.WithContext(context.WithValue(
+			req.Context(),
+			contextKey("yotiError"),
+			fmt.Sprintf("%s", err),
+		)))
+	}
+
+	share, err := client.CreateShareURL(&scenario)
+	if err != nil {
+		errorPage(w, req.WithContext(context.WithValue(
+			req.Context(),
+			contextKey("yotiError"),
+			fmt.Sprintf("%s", err.Error()),
+		)))
+		return
+	}
+
+	templateVars := map[string]interface{}{
```

Comment:
why the map string interface, could you use a known struct for this?

Reply:
I'm just moving this file, but it does appear in all our templates. 

I don't think it's worth creating a struct for these, since we would have to create 4/5 different structs which would all be used once,  and we don't gain the benefits of using a struct when we reference this object in the template: we just do e.g. `{{.pageTitle}}` (with no autocomplete), and the template rendering engine picks it up at runtime